packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -500,6 +545,126 @@ class FirestoreListView<Document> extends FirestoreQueryBuilder<Document> {
+    Axis scrollDirection = Axis.vertical,
+    bool showLoadingIndicator = false,
+    bool reverse = false,
+    ScrollController? controller,
+    bool? primary,
+    ScrollPhysics? physics,
+    bool shrinkWrap = false,
+    EdgeInsetsGeometry? padding,
+    ChildIndexGetter? findChildIndexCallback,
+    bool addAutomaticKeepAlives = true,
+    bool addRepaintBoundaries = true,
+    bool addSemanticIndexes = true,
+    double? cacheExtent,
+    DragStartBehavior dragStartBehavior = DragStartBehavior.start,
+    ScrollViewKeyboardDismissBehavior keyboardDismissBehavior =
+        ScrollViewKeyboardDismissBehavior.manual,
+    String? restorationId,
+    Clip clipBehavior = Clip.hardEdge,
+  }) : super(
+          key: key,
+          query: query,
+          pageSize: pageSize ?? (showLoadingIndicator ? 20 : 10),
+          builder: (context, snapshot, _) {
+            if (snapshot.isFetching) {
+              return loadingBuilder?.call(context) ??
+                  const Center(child: CircularProgressIndicator());
+            }
+
+            if (snapshot.hasError && errorBuilder != null) {
+              return errorBuilder(
+                context,
+                snapshot.error!,
+                snapshot.stackTrace!,
+              );
+            }
+
+            if (snapshot.docs.isEmpty && emptyBuilder != null) {
+              return emptyBuilder(context);
+            }
+
+            final itemCount = snapshot.docs.length;
+
+            return ListView.separated(
+              itemCount: itemCount,
+              itemBuilder: (context, index) {
+                final isLastItem = index + 1 == itemCount;
+                if (!showLoadingIndicator && isLastItem && snapshot.hasMore) {
+                  snapshot.fetchMore();
+                }
+
+                final doc = snapshot.docs[index];
+                return showLoadingIndicator
+                    ? BuildAwareWidget(
+                        onBuild: () async {
+                          /// if we reached the end of the currently obtained
+                          /// items, we try to obtain more items
+                          if (isLastItem && snapshot.hasMore) {
+                            /// We wait just so we get to see the progress
+                            /// indicator for a while.
+                            await Future.delayed(
+                              const Duration(milliseconds: 800),
+                              () => snapshot.fetchMore(),
+                            );
+                          }
+                        },
+                        child: Column(
+                          crossAxisAlignment: CrossAxisAlignment.start,
+                          children: [
+                            itemBuilder(context, doc),
+                            if (isLastItem && snapshot.hasMore)
+                              loadingIndicatorBuilder?.call(context) ??
+                                  const Padding(
```

Comment:
there is a [`LoadingIndicator`](https://pub.dev/documentation/firebase_ui_shared/latest/firebase_ui_shared/LoadingIndicator-class.html) in `firebase_ui_shared`